server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -393,6 +393,12 @@ public class Node implements Closeable {
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> NODE_SEARCH_CACHE_OVERFLOW_SETTING = Setting.boolSetting(
+        "node.search.cache.overflow",
+        false,
```

Comment:
I think the overflow behavior is unquestionably bad, and probably better described as a bug. The behavior change is really a bug fix. I wonder if we even need the setting?

Reply:
Removed node setting as well as some unit tests for cache overflow. It seemed like there was a clear intention for this cache to overflow when it was initially added so I was hesitant to disable it completely.